Tank.java

```diff
@@ -1 +1,459 @@
+	}
+
+	void move() {
+
+		this.oldX = x;
+		this.oldY = y;
+
+		switch (direction) {
+			case L:
+				x -= speedX;
+				break;
+			case U:
+				y -= speedY;
+				break;
+			case R:
+				x += speedX;
+				break;
+			case D:
+				y += speedY;
+				break;
+			case STOP:
+				break;
+		}
+
+		if (this.direction != Direction.STOP) {
+			this.myDirection = this.direction;
+		}
+
+		if (x < 0)
+			x = 0;
+		if (y < 40)
+			y = 40;
+		if (x + Tank.WIDTH > TankClient.Fram_width)
+			x = TankClient.Fram_width - Tank.WIDTH;
+		if (y + Tank.LENGHT > TankClient.Fram_length)
+			y = TankClient.Fram_length - Tank.LENGHT;
+
+		if (!good) {
+			Direction[] directons = Direction.values();
+			if (step == 0) {
+				step = r.nextInt(12) + 3;
+				int mod=r.nextInt(9);
+				if (playertankaround()){
+					if(x==tc.homeTank.x){ if(y>tc.homeTank.y) direction=directons[1];
+					else if (y<tc.homeTank.y) direction=directons[3];
+					}else if(y==tc.homeTank.y){ if(x>tc.homeTank.x) direction=directons[0];
+					else if (x<tc.homeTank.x) direction=directons[2];
+					}
+					else{
+						int rn = r.nextInt(directons.length);
+						direction = directons[rn];
+					}
+					rate=2;
+				}else if (mod==1){
+					rate=1;
+				}else if(1<mod&&mod<=3){
+					rate=1;
+				}else{
+					int rn = r.nextInt(directons.length);
+					direction = directons[rn];
+					rate=1;}
+			}
+			step--;
+			if(rate==2){
+				if (r.nextInt(40) > 35)
+					this.fire();
+			}else if (r.nextInt(40) > 38)
+				this.fire();
+		}
+	}
+	public boolean playertankaround(){
+		int rx=x-15;
+		int ry=y-15;
+		if((x-15)<0) rx=0;
+		if((y-15)<0)ry=0;
+		Rectangle a=new Rectangle(rx, ry,60,60);
+		if (this.live && a.intersects(tc.homeTank.getRect())) {
+			return true;
+		}
+		return false;
+	}
+	public int getzone(int x,int y){
+		int tempx=x;
+		int tempy=y;
+		if (tempx<85&&tempy<300) return 11;
+		else if(tempx>85&&tempx<140&&tempy>0&&tempy<100) return 9;
+		else if(tempx>85&&tempx<140&&tempy>254&&tempy<300) return 10;
+		else if(tempx>0&&tempx<200&&tempy>300&&tempy<715) return 12;
+		else if(tempx>140&&tempx<400&&tempy>0&&tempy<150) return 7;
+		else if(tempx>140&&tempx<400&&tempy>210&&tempy<300) return 8;
+		else if(tempx>400&&tempx<500&&tempy>0&&tempy<300) return 6;
+		else if(tempx>500&&tempy>0&&tempy<180) return 5;
+		else if(tempx>500&&tempy>180&&tempy<300) return 4;
+		else if(tempx>520&&tempx<600&&tempy>3000&&tempy<715) return 2;
+		else if(tempx>600&&tempy>300&&tempy<715) return 3;
+		return 1;
+	}
+	public int getDirect(int b){
+		return 4;
+	}
+	private void changToOldDir() {
+		x = oldX;
+		y = oldY;
+	}
+
+	public void keyPressed(KeyEvent e) {
+		int key = e.getKeyCode();
+		if (player==1){
+			switch (key) {
+				case KeyEvent.VK_R:
+					tc.tanks.clear();
+					tc.bullets.clear();
+					tc.trees.clear();
+					tc.otherWall.clear();
+					tc.homeWall.clear();
+					tc.metalWall.clear();
+					tc.homeTank.setLive(false);
+					if (tc.tanks.size() == 0) {
+						for (int i = 0; i < 20; i++) {
```

Comment:
Panašu, kad čia inicialiuojami priešų tankai. 
Čia galima būtų refaktorinti, extract method for ciklą.
Tai pagerintų skaitomumą.